rpc/src/module/ipc.rs

```diff
@@ -0,0 +1,883 @@
+            .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?;
+        let mut meta = tx.resolved_cell_deps[ci].clone();
+        meta.mem_cell_data = Some(ipc_script_data);
+        dl.cell
+            .insert(meta.out_point.clone(), CellStatus::Live(meta));
+        return Ok(env);
+    }
+
+    fn ipc_env(
+        &self,
+        dl: &mut Resource,
+        ipc_script_data: Bytes,
+        env: Option<IpcEnv>,
+    ) -> Result<IpcEnv> {
+        // If the user doesn't pass in an env, we will construct a default env.
+        match env {
+            Some(val) => self.ipc_env_some(dl, ipc_script_data, val),
+            None => self.ipc_env_none(dl, ipc_script_data),
+        }
+    }
+
+    fn ipc_script_data_by_out_point(&self, script_locator: IpcScriptLocator) -> Result<Bytes> {
+        let out_point = script_locator
+            .out_point
+            .ok_or_else(|| RPCError::custom_with_error(RPCError::IPC, "Unexpected"))?;
+        let out_point: packed::OutPoint = out_point.into();
+        if let Some((data, _)) = self.shared.snapshot().get_cell_data(&out_point) {
+            Ok(data)
+        } else {
+            Err(RPCError::custom(
+                RPCError::IPC,
+                format!("Get out point failed: {}", out_point),
+            ))
+        }
+    }
+
+    fn ipc_script_data_by_type_id_args(&self, script_locator: IpcScriptLocator) -> Result<Bytes> {
+        match &self.indexer_rpc_impl {
+            Some(data) => {
+                let type_id_args = script_locator
+                    .type_id_args
+                    .ok_or_else(|| RPCError::custom_with_error(RPCError::IPC, "Unexpected"))?;
+                let search_key = IndexerSearchKey {
+                    script: Script {
+                        code_hash: TYPE_ID_CODE_HASH,
+                        hash_type: ScriptHashType::Type,
+                        args: JsonBytes::from_vec(type_id_args.as_bytes().to_vec()),
+                    },
+                    script_type: IndexerScriptType::Type,
+                    ..Default::default()
+                };
+                let object_vec = data
+                    .handle
+                    .get_cells(search_key, IndexerOrder::Desc, 1.into(), None)
+                    .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?
+                    .objects;
+                Ok(object_vec
+                    .first()
+                    .ok_or_else(|| {
+                        RPCError::custom(
+                            RPCError::IPC,
+                            format!("Get type id args failed: {}", type_id_args),
+                        )
+                    })?
+                    .output_data
+                    .clone()
+                    .ok_or_else(|| RPCError::custom(RPCError::IPC, "Unexpected"))
+                    .map(|e| e.into_bytes())?)
+            }
+            None => Err(RPCError::custom(
+                RPCError::IPCIndexerIsDisabled,
```

Comment:
seems from the code, if IPC module is not enabled, `ipc_script_data_by_type_id_args` will never be called here. 
it's not necessary to add a RPCError::IPCIndexerIsDisabled code for it.

Reply:
never mind, it's not IPC module here.

Reply:
so if IPC module is enabled, it will require Index module required, could we just validate this rule when parsed config file.
otherwise user can only figure out this rule when error happened.

Reply:
When the `script_locator` parameter of ipc_call is `type_id_args`, it will depend on indexer;
When it is `out_point`, it will not depend on indexer.

So it can only be checked at runtime.